### ISSUE TITLE
Incorrect error TS2339: Property 'kind' does not exist on type 'Resource'

### ISSUE THREAD

**Issue by nagapavan** [palantir/tslint#3927]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: (pick one) CLI and gulp-tslint

#### TypeScript code being linted

```ts
      const api = this.kubernetesApi(namespace, clusterInfo);
      api.group(manifest).ns(namespace).kind(manifest).post({ body: manifest }).then((result: any) => {
        resolve(result);
      }).catch((error: any) => {
        reject(error);
      });
```

[Related index.d.ts snippet: ](https://github.com/godaddy/kubernetes-client/blob/5.3.0/typings/index.d.ts)
```ts
  interface Resource extends ApiGroup {
    (resourceName: string) : Resource;
  }

  interface NamespacesResource extends Resource {
    kind(k: { kind: string } | string): ApiGroup;
  }

interface ApiGroup {
// ...
namespaces?: NamespacesResource;
ns?: NamespacesResource;
// ...
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended",
    "tslint-sonarts"
  ],
  "rules": {
    "interface-name": false,
    "no-namespace": false,
    "align": false,
    "no-identical-functions": false,
    "cognitive-complexity": false,
    "no-commented-code": false,
    "no-var-requires": false,
    "prefer-immediate-return": false,
    "no-dead-store": false,
    "typedef-whitespace": false,
    "object-literal-sort-keys": false
  }
}
```

#### Actual behavior

> Semantic Error:  error TS2339: Property 'kind' does not exist on type 'Resource'

#### Expected behavior
No error (semantic/actual) or warning should have been generated, since `ns`/`namespaces` is an instance of `NamespacesResource` extended from `Resource` - which has the method `kind()`

**Comment by giladgray** [comment]
@nagapavan this is a typescript compiler error (TS####), not a tslint error. are you sure it's related to this project?

**Comment by nagapavan** [comment]
@giladgray apologies. Will move this issue to the TypeScript repo.
